plugins/che-plugin-analytics/che-plugin-analytics-wsmaster/src/main/java/com/redhat/che/plugin/analytics/wsmaster/AnalyticsService.java

```diff
@@ -43,4 +50,29 @@ public String segmentWriteKey() {
   public String woopraDomain() {
     return woopraDomain == null ? "" : woopraDomain;
   }
+
+  private String getLog(String message, HttpHeaders headers) {
+    StringBuffer log = new StringBuffer("[E2E Registration Flow");
+
+    String ip = headers.getHeaderString("X-Forwarded-For");
+    if (ip != null) {
+      log.append("- IP = " + ip);
+    }
+    log.append("] ").append(message);
+    return log.toString();
```

Comment:
maybe a string format is faster ?

Reply:
tried to make it nicer. Is it more understandable like this ?